plugins/s3_auth/aws_auth_v4.cc

```diff
@@ -0,0 +1,691 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+/**
+ * @file aws_auth_v4.cc
+ * @brief AWS Auth v4 signing utility.
+ * @see aws_auth_v4.h
+ */
+
+#include <cstring>        /* strlen() */
+#include <ctime>          /* strftime(), time(), gmtime_r() */
+#include <iomanip>        /* std::setw */
+#include <sstream>        /* std::stringstream */
+#include <openssl/sha.h>  /* SHA(), sha256_Update(), SHA256_Final, etc. */
+#include <openssl/hmac.h> /* HMAC() */
+
+#undef AWS_AUTH_V4_DETAILED_DEBUG_OUTPUT
+#ifdef AWS_AUTH_V4_DETAILED_DEBUG_OUTPUT
```

Comment:
@SolidWallOfCode: used it only for manually enable sending extra-detailed-debugging to `stdout` if necessary while developing/testing but did not want such details while running the plugin or when running the unit-tests at all times (there might be a better way to achieve the same)